src/lib.rs

```diff
@@ -547,6 +714,131 @@ impl Sender {
 
         Ok(true)
     }
+
+    /// Send the specified message, waiting for sufficient contiguous space to become available in the ring buffer
+    /// if necessary. Uses the lower level `ShmWriter` trait to write messages directly into the mmap buffer.
+    ///
+    /// The size of the message must be greater than zero or else this method will return
+    /// `Err(`[`Error::ZeroSizedMessage`](enum.Error.html#variant.ZeroSizedMessage)`))`.  If the message size is
+    /// greater than the ring buffer capacity, this method will return
+    /// `Err(`[`Error::MessageTooLarge`](enum.Error.html#variant.MessageTooLarge)`))`.
+    pub fn send_direct(&self, value: &impl ShmWriter) -> Result<()> {
```

Comment:
I think all the `send_direct*` methods should be marked `unsafe` given the inherent unsafety of `ShmWriter`.

Reply:
I agree, can easily update those